base/special/trig.jl

```diff
@@ -197,7 +197,9 @@ function sincos(x::T) where T<:Union{Float32, Float64}
         return -co, si
     end
 end
-sincos(x::Real) = sincos(float(x))
+sincos(x::T) where {T <: Union{Integer, Rational}} = sincos(float(x))
```

Comment:
What about irrational?

Reply:
Yes, but irrationals don't use the faster version, right? (I'm on my phone, I can't check myself right now)

Reply:
Oh, my bad, of course you're right, it should be `{Integer, Rational, Irrational}`. This changes it such that `sincos(pi)` is now slower than before.

Reply:
Ah, good point, thanks, I'll change it.

I don't understand how to interpret the test errors -- can anybody help me?